example-get-started-experiments/code/sagemaker/deploy_model.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import re
+import sys
+
+from sagemaker.deserializers import JSONDeserializer
+from sagemaker.pytorch import PyTorchModel
+from sagemaker.serverless import ServerlessInferenceConfig
+
+
+memory_size = { 
+    "dev": 4096 ,
+    "staging": 4096,
+    "prod": 6144 ,
+    "default": 4096,
+}
+max_concurrency = { 
+    "dev": 5,
+    "staging": 5,
+    "prod": 10,
+    "default": 5,
+}
+
+
+def deploy(
+    name: str,
+    stage: str,
+    version: str,
+    model_data: str,
+    role: str,
+):
+    sagemaker_logger = logging.getLogger("sagemaker")
+    sagemaker_logger.setLevel(logging.DEBUG)
+    sagemaker_logger.addHandler(logging.StreamHandler(sys.stdout))
+
+    composed_name =  re.sub(
+        r"[^a-zA-Z0-9\-]", "-", f"{name}-{version}-{stage}")
+    
+    model = PyTorchModel(
+        name=composed_name,
+        model_data=model_data,
+        framework_version="1.12",
+        py_version="py38",
+        role=role,
+        env={
+            "SAGEMAKER_MODEL_SERVER_TIMEOUT": "3600",
+            "TS_MAX_RESPONSE_SIZE": "2000000000",
+            "TS_MAX_REQUEST_SIZE": "2000000000",
+            "MMS_MAX_RESPONSE_SIZE": "2000000000",
+            "MMS_MAX_REQUEST_SIZE": "2000000000",
+        },
+    )
+
+
+    return model.deploy(
```

Comment:
> 

> @daavoo If I am trying to deploy a new model version to an existing stage, does this code create a new endpoint or does it update the existing endpoint? (I need to confirm this for the blog post)

Not sure I understand the first part but a new endpoint is created from combining the name, version, and stage, so any change to those 3 things result in a new endpoint

Reply:
So if I deploy v1 to prod today, and v2 to prod tomorrow, will I have 2 different endpoints with 2 different endpoint names? Shouldn't the prod endpoint always be the same? (just like studio prod version is always studio.iterative.ai) So that any clients running inference against the prod endpoint don't have to update their endpoint addresses each time there's a new deployment.

Reply:
The version shouldn't be part of the endpoint name, right?

Reply:
> Shouldn't the prod endpoint always be the same? 

I think it is a matter of opinion and depends on the pattern we expect/want to showcase:

A) The endpoint is directly queried by external consumers
B) There is some app&url that uses the right endpoint internally without exposing it to external consumers.

> (just like studio prod version is always studio.iterative.ai) So that any clients running inference against the prod endpoint don't have to update their endpoint addresses each time there's a new deployment.

That is the user-facing app&url and it only "points" to the correct software because there are gitops bumping the internal studio version (i.e. https://github.com/iterative/itops/pull/2238)

Reply:
So then we should also create a service to delete stale endpoints, right? Else, we will end up with a bunch of endpoints over time.

Reply:
> So then we should also create a service to delete stale endpoints, right? Else, we will end up with a bunch of endpoints over time.

I would expect that to be configured / handled on the AWS side.

Anyhow, as per your suggestions, I will send a new P.R. creating/updating a single endpoint per stage